controllers/workspace/devworkspace_controller.go

```diff
@@ -125,6 +125,12 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 
 	// Stop failed workspaces
 	if workspace.Status.Phase == dw.DevWorkspaceStatusFailed && workspace.Spec.Started {
+		// If debug annotation is present, leave the deployment in place to let users
```

Comment:
leaving the deployment in place is the result, but what actually it does - do not initialize stopping of failed/[failing](https://github.com/amisevsk/devworkspace-operator/blob/start-failed-workspace/controllers/workspace/devworkspace_controller.go#L127) workspace.

Do I understand correctly, that with https://github.com/devfile/devworkspace-operator/pull/424/files users are supposed to manually stop workspace before starting it again if debug mode is activated?

Reply:
That would be the requirement, yes -- it leaves failed workspaces with `spec.started=true` so that you can debug issues (basically it treats workspaces with the annotation similar to how they were treated before https://github.com/devfile/devworkspace-operator/pull/362).

The other way to implement this would be to leave the deployment in place for stopped workspaces when the debug annotation is present, which I think is more confusing.

This PR will need to be reworked slightly once #424 is merged.